packages/pn-commons/src/utils/notification.utility.ts

```diff
@@ -1,3 +1,5 @@
+/* eslint-disable sonarjs/cognitive-complexity */
```

Comment:
I refuse to do this. There are two functions that (in Sonar' opinion) have a too great complexity
- `getLegalFactLabel` which I didn't touch at all in this PR.
- `populateMacroSteps`, which complexity jumped to 46 because ... of the modifications you suggest in your other comments.